diff --git a/cmd/prometheus/main.go b/cmd/prometheus/main.go
index 864799cf5..88283db40 100644
--- a/cmd/prometheus/main.go
+++ b/cmd/prometheus/main.go
@@ -55,9 +55,6 @@ import (
 	toolkit_web "github.com/prometheus/exporter-toolkit/web"
 	"go.uber.org/atomic"
 	"go.uber.org/automaxprocs/maxprocs"
-	"k8s.io/client-go/rest"
-	"k8s.io/klog"
-	klogv2 "k8s.io/klog/v2"
 
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
@@ -106,24 +103,6 @@ type klogv1Writer struct{}
 // by klog's upstream klogv1/v2 coexistence example:
 // https://github.com/kubernetes/klog/blob/main/examples/coexist_klog_v1_and_v2/coexist_klog_v1_and_v2.go
 func (klogv1Writer) Write(p []byte) (n int, err error) {
-	if len(p) < klogv1DefaultPrefixLength {
-		klogv2.InfoDepth(klogv1OutputCallDepth, string(p))
-		return len(p), nil
-	}
-
-	switch p[0] {
-	case 'I':
-		klogv2.InfoDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	case 'W':
-		klogv2.WarningDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	case 'E':
-		klogv2.ErrorDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	case 'F':
-		klogv2.FatalDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	default:
-		klogv2.InfoDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	}
-
 	return len(p), nil
 }
 
@@ -835,12 +814,6 @@ func main() {
 	noStepSubqueryInterval := &safePromQLNoStepSubqueryInterval{}
 	noStepSubqueryInterval.Set(config.DefaultGlobalConfig.EvaluationInterval)
 
-	klogv2.SetSlogLogger(logger.With("component", "k8s_client_runtime"))
-	klog.SetOutputBySeverity("INFO", klogv1Writer{})
-	// Avoid duplicate API deprecation warnings (e.g., "v1 Endpoints is deprecated in v1.33+...")
-	// that can pollute the logs.
-	rest.SetDefaultWarningHandlerWithContext(logging.NewDedupDeprecationWarningLogger())
-
 	modeAppName := "Prometheus Server"
 	mode := "server"
 	if agentMode {
@@ -885,17 +858,6 @@ func main() {
 		discoveryManagerNotify  *discovery.Manager
 	)
 
-	// Kubernetes client metrics are used by Kubernetes SD.
-	// They are registered here in the main function, because SD mechanisms
-	// can only register metrics specific to a SD instance.
-	// Kubernetes client metrics are the same for the whole process -
-	// they are not specific to an SD instance.
-	err = discovery.RegisterK8sClientMetricsWithPrometheus(prometheus.DefaultRegisterer)
-	if err != nil {
-		logger.Error("failed to register Kubernetes client metrics", "err", err)
-		os.Exit(1)
-	}
-
 	sdMetrics, err := discovery.CreateAndRegisterSDMetrics(prometheus.DefaultRegisterer)
 	if err != nil {
 		logger.Error("failed to register service discovery metrics", "err", err)
diff --git a/cmd/promtool/main.go b/cmd/promtool/main.go
index 6555ccc6f..b22e5bfcb 100644
--- a/cmd/promtool/main.go
+++ b/cmd/promtool/main.go
@@ -48,7 +48,6 @@ import (
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/discovery/file"
-	"github.com/prometheus/prometheus/discovery/kubernetes"
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/model/rulefmt"
@@ -712,10 +711,6 @@ func checkConfig(agentMode bool, filename string, checkSyntaxOnly bool) ([]strin
 
 		for _, c := range scfg.ServiceDiscoveryConfigs {
 			switch c := c.(type) {
-			case *kubernetes.SDConfig:
-				if err := checkTLSConfig(c.HTTPClientConfig.TLSConfig, checkSyntaxOnly); err != nil {
-					return nil, nil, err
-				}
 			case *file.SDConfig:
 				if checkSyntaxOnly {
 					break
