diff --git a/cmd/prometheus/main.go b/cmd/prometheus/main.go
index 864799cf5..88283db40 100644
--- a/cmd/prometheus/main.go
+++ b/cmd/prometheus/main.go
@@ -55,9 +55,6 @@ import (
 	toolkit_web "github.com/prometheus/exporter-toolkit/web"
 	"go.uber.org/atomic"
 	"go.uber.org/automaxprocs/maxprocs"
-	"k8s.io/client-go/rest"
-	"k8s.io/klog"
-	klogv2 "k8s.io/klog/v2"
 
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
@@ -106,24 +103,6 @@ type klogv1Writer struct{}
 // by klog's upstream klogv1/v2 coexistence example:
 // https://github.com/kubernetes/klog/blob/main/examples/coexist_klog_v1_and_v2/coexist_klog_v1_and_v2.go
 func (klogv1Writer) Write(p []byte) (n int, err error) {
-	if len(p) < klogv1DefaultPrefixLength {
-		klogv2.InfoDepth(klogv1OutputCallDepth, string(p))
-		return len(p), nil
-	}
-
-	switch p[0] {
-	case 'I':
-		klogv2.InfoDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	case 'W':
-		klogv2.WarningDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	case 'E':
-		klogv2.ErrorDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	case 'F':
-		klogv2.FatalDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	default:
-		klogv2.InfoDepth(klogv1OutputCallDepth, string(p[klogv1DefaultPrefixLength:]))
-	}
-
 	return len(p), nil
 }
 
@@ -835,12 +814,6 @@ func main() {
 	noStepSubqueryInterval := &safePromQLNoStepSubqueryInterval{}
 	noStepSubqueryInterval.Set(config.DefaultGlobalConfig.EvaluationInterval)
 
-	klogv2.SetSlogLogger(logger.With("component", "k8s_client_runtime"))
-	klog.SetOutputBySeverity("INFO", klogv1Writer{})
-	// Avoid duplicate API deprecation warnings (e.g., "v1 Endpoints is deprecated in v1.33+...")
-	// that can pollute the logs.
-	rest.SetDefaultWarningHandlerWithContext(logging.NewDedupDeprecationWarningLogger())
-
 	modeAppName := "Prometheus Server"
 	mode := "server"
 	if agentMode {
@@ -885,17 +858,6 @@ func main() {
 		discoveryManagerNotify  *discovery.Manager
 	)
 
-	// Kubernetes client metrics are used by Kubernetes SD.
-	// They are registered here in the main function, because SD mechanisms
-	// can only register metrics specific to a SD instance.
-	// Kubernetes client metrics are the same for the whole process -
-	// they are not specific to an SD instance.
-	err = discovery.RegisterK8sClientMetricsWithPrometheus(prometheus.DefaultRegisterer)
-	if err != nil {
-		logger.Error("failed to register Kubernetes client metrics", "err", err)
-		os.Exit(1)
-	}
-
 	sdMetrics, err := discovery.CreateAndRegisterSDMetrics(prometheus.DefaultRegisterer)
 	if err != nil {
 		logger.Error("failed to register service discovery metrics", "err", err)
diff --git a/cmd/promtool/main.go b/cmd/promtool/main.go
index 6555ccc6f..b22e5bfcb 100644
--- a/cmd/promtool/main.go
+++ b/cmd/promtool/main.go
@@ -48,7 +48,6 @@ import (
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/discovery/file"
-	"github.com/prometheus/prometheus/discovery/kubernetes"
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/model/rulefmt"
@@ -712,10 +711,6 @@ func checkConfig(agentMode bool, filename string, checkSyntaxOnly bool) ([]strin
 
 		for _, c := range scfg.ServiceDiscoveryConfigs {
 			switch c := c.(type) {
-			case *kubernetes.SDConfig:
-				if err := checkTLSConfig(c.HTTPClientConfig.TLSConfig, checkSyntaxOnly); err != nil {
-					return nil, nil, err
-				}
 			case *file.SDConfig:
 				if checkSyntaxOnly {
 					break
diff --git a/util/logging/dedupe.go b/util/logging/dedupe.go
index 9a74278a7..244cd6495 100644
--- a/util/logging/dedupe.go
+++ b/util/logging/dedupe.go
@@ -18,9 +18,6 @@ import (
 	"log/slog"
 	"sync"
 	"time"
-
-	"github.com/grafana/regexp"
-	"k8s.io/client-go/rest"
 )
 
 const (
@@ -137,44 +134,3 @@ func (d *Deduper) run() {
 		}
 	}
 }
-
-// deprecationRegex matches the format of Kubernetes API deprecation warnings:
-// See https://github.com/kubernetes/kubernetes/blob/da663405beb487d66c27a0220ea4073305ae9077/staging/src/k8s.io/apiserver/pkg/endpoints/deprecation/deprecation.go#L117.
-var deprecationRegex = regexp.MustCompile(`\S+ \S+ is deprecated in v\d+\.\d+\+`)
-
-// Even though deprecation warnings should be bounded in number, this safeguard should help prevent leaks.
-const maxDeprecationWarnings = 32
-
-// DedupDeprecationWarningLogger deduplicates Kube API deprecation warnings by message before logging them.
-// Inspired by https://github.com/kubernetes/kubernetes/blob/3edae6c1c49958fd10a708d9cc8c4c9e7f5fb6e8/staging/src/k8s.io/client-go/rest/warnings.go#L113
-type DedupDeprecationWarningLogger struct {
-	logger rest.WarningHandlerWithContext
-	lock   sync.Mutex
-	logged map[string]struct{}
-}
-
-func NewDedupDeprecationWarningLogger() *DedupDeprecationWarningLogger {
-	return &DedupDeprecationWarningLogger{
-		logger: rest.WarningLogger{},
-		logged: make(map[string]struct{}),
-	}
-}
-
-func (w *DedupDeprecationWarningLogger) HandleWarningHeaderWithContext(ctx context.Context, code int, agent, message string) {
-	if code != 299 || message == "" {
-		return
-	}
-
-	w.lock.Lock()
-	defer w.lock.Unlock()
-
-	if _, seen := w.logged[message]; seen {
-		return
-	}
-
-	if deprecationRegex.MatchString(message) && len(w.logged) < maxDeprecationWarnings {
-		w.logged[message] = struct{}{}
-	}
-
-	w.logger.HandleWarningHeaderWithContext(ctx, code, agent, message)
-}
